recipes/gettext/all/conanfile.py

```diff
@@ -0,0 +1,127 @@
+    homepage = "https://www.gnu.org/software/gettext"
+    license = "GPL-3.0-or-later"
+    settings = "os_build", "arch_build", "arch", "compiler", "build_type"
+    exports_sources = ["patches/*.patch"]
+
+    
+
+    requires = ("libiconv/1.15",
+                "libxml2/2.9.9")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _gettext_folder(self):
+        return "gettext-runtime"
+
+    @property
+    def _make_args(self):
+        return None
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            if "CONAN_BASH_PATH" not in os.environ:
+                self.build_requires("msys2/20190524")
+        if self._is_msvc:
+            self.build_requires("automake/1.16.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "gettext-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)    
+        libiconv_prefix = self.deps_cpp_info["libiconv"].rootpath
+        libxml2_prefix = self.deps_cpp_info["libxml2"].rootpath
+        libiconv_prefix = tools.unix_path(libiconv_prefix) if tools.os_info.is_windows else libiconv_prefix
+        libxml2_prefix = tools.unix_path(libxml2_prefix) if tools.os_info.is_windows else libxml2_prefix
+        args = ["HELP2MAN=/bin/true",
+                "EMACS=no",
+                "--disable-nls",
+                "--disable-dependency-tracking",
+                "--enable-relocatable",
+                "--disable-c++",
+                "--disable-java",
+                "--disable-csharp",
+                "--disable-libasprintf",
+                "--disable-curses",
+                "--with-libiconv-prefix=%s" % libiconv_prefix,
+                "--with-libxml2-prefix=%s" % libxml2_prefix]
+        build = None
+        host = None
+        rc = None
+        if self.options.get_safe("shared"):
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--enable-static"])
+        if self._is_msvc:
+            # INSTALL.windows: Native binaries, built using the MS Visual C/C++ tool chain.
+            build = False
+            if self.settings.arch == "x86":
+                host = "i686-w64-mingw32"
+                rc = "windres --target=pe-i386"
+            elif self.settings.arch == "x86_64":
+                host = "x86_64-w64-mingw32"
+                rc = "windres --target=pe-x86-64"
+            automake_perldir = os.getenv('AUTOMAKE_PERLLIBDIR')
+            if automake_perldir.startswith('/mnt/'):
```

Comment:
Why remove this prefix? Will the script still work?

Reply:
On my machine it only works like this, and it does not seem to break on CCI.

Reply:
Can you run the test_package of automake? `conan test recipes/automake/all/test_package automake/1.16.1@ -tbf /tmp/automake_test_builddir`

Reply:
it fails:
```
automake/1.16.1: Already installed!
automake/1.16.1: Appending PATH env var with : c:\users\eric\.conan\data\automake\1.16.1\_\_\package\01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4\bin
automake/1.16.1: Setting ACLOCAL to /mnt/c/Users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin/aclocal.exe
automake/1.16.1: Setting AUTOMAKE_DATADIR to /mnt/c/Users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin/share
automake/1.16.1: Setting AUTOMAKE_LIBDIR to /mnt/c/Users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin/share/automake-1.16
automake/1.16.1: Setting AUTOMAKE_PERLLIBDIR to /mnt/c/Users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin/share/automake-1.16
automake/1.16.1: Setting AUTOMAKE to /mnt/c/Users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin/automake.exe
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
automake/1.16.1 (test package): Applying build-requirement: cmake_installer/3.15.5@conan/stable
automake/1.16.1 (test package): Applying build-requirement: msys2/20190524
automake/1.16.1 (test package): Generator txt created conanbuildinfo.txt
automake/1.16.1 (test package): Generated conaninfo.txt
automake/1.16.1 (test package): Generated graphinfo
Using lockfile: 'C:\tmp\automake_test_builddir/conan.lock'
Using cached profile from lockfile
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
automake/1.16.1 (test package): Running build()
automake/1.16.1 (test package): run_in_windows_bash: c:\.conan\d8a07f\1\bin\usr\bin\bash.exe --login -c ^"cd \^"/c/tmp/automake_test_builddir\^" ^&^& PATH=\^"/c/users/eric/.conan/data/cmake_installer/3.15.5/conan/stable/package/60b04a80ac4dd5c23f1acffacc679e9902e636e5/bin:/c/.conan/d8a07f/1/bin/usr/bin:/c/users/eric/.conan/data/automake/1.16.1/_/_/package/01ebd508159eeb2c18efe7e3c3a025c7c52c6ee4/bin:/c/users/eric/.conan/data/autoconf/2.69/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin:$PATH\^" ^&^& /mnt/c/users/eric/.conan/data/autoconf/2.69/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin/autoreconf.exe --verbose --install --force ^"        automake/1.16.1 (test package): /usr/bin/bash: /mnt/c/users/eric/.conan/data/autoconf/2.69/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin/autoreconf.exe: No such file or directory
ERROR: automake/1.16.1 (test package): Error in build() method, line 28
        self.run("{} --verbose --install --force".format(os.environ["AUTORECONF"]), win_bash=tools.os_info.is_windows)
        ConanException: Error 127 while executing /mnt/c/users/eric/.conan/data/autoconf/2.69/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin/autoreconf.exe --verbose --install --force
```
I'm running conan from windows console, not msys2 bash

Reply:
It looks like conan has detected `path_flavor==tools.WSL` (see `conans.client.tools.win.unix_path`)
Do you have any idea why this is so?
Guess I'll have to modify the autoconf/automake/libtool scripts to `tools.unix_path(..., path_flavor=toolsMSYS2)`

Reply:
I guess it detected WSL because it is the first (and only) bash available in path (msys2 package is not reachable in path from automake's `package_info`). My take on this is that you should not use `tools.unix_path` in automake's package_info. Consumers will do it if needed, based on the actual bash they use.

Reply:
That is a problem: `AUTOMAKE_PERLLIBDIR` is a variable required by automake but its value is dependent on the subsystem it is running in. But that value is set inside `package_info` where it can only guess what subsystem to use.
@uilianries @danimtb Suggestions?

Reply:
@ericLemanissier I have pushed a new branch to my fork at https://github.com/madebr/conan-center-index/tree/autotools_win_subsystem.
Can you test it?
You'll need to build autoconf and automake:
```
conan create recipes/autoconf/all autoconf/2.69@
conan create recipes/automake/all automake/1.16.1@
```
I have added a new `path_flavor` option that is initialized to `tools.os_info.detect_windows_subsystem()` but can be overridden.